com.unity.render-pipelines.core/Editor/Lighting/GUIFramework/GUIState.cs

```diff
@@ -0,0 +1,257 @@
+        /// Add Control
+        /// </summary>
+        /// <param name="controlID">Control ID</param>
+        /// <param name="distance">The distance</param>
+        public void AddControl(int controlID, float distance)
+        {
+            HandleUtility.AddControl(controlID, distance);
+        }
+
+        /// <summary>
+        /// Slider
+        /// </summary>
+        /// <param name="id">ID</param>
+        /// <param name="sliderData">Slider Data</param>
+        /// <param name="newPosition">New position</param>
+        /// <returns>true if changed</returns>
+        public bool Slider(int id, SliderData sliderData, out Vector3 newPosition)
+        {
+            if (mouseButton == 0 && eventType == EventType.MouseDown)
+            {
+                hotControl = 0;
+                nearestControl = id;
+            }
+
+            EditorGUI.BeginChangeCheck();
+            newPosition = Handles.Slider2D(id, sliderData.position, sliderData.forward, sliderData.right, sliderData.up, 1f, nullCap, Vector2.zero);
+            return EditorGUI.EndChangeCheck();
+        }
+
+        /// <summary>
+        /// Use event
+        /// </summary>
+        public void UseEvent()
+        {
+            Event.current.Use();
+        }
+
+        /// <summary>
+        /// Calls the methods in its invocation list when repaint
+        /// </summary>
+        public void Repaint()
+        {
+            HandleUtility.Repaint();
+        }
+
+        /// <summary>
+        /// Test if has current camera
+        /// </summary>
+        /// <returns></returns>
+        public bool HasCurrentCamera()
+        {
+            return Camera.current != null;
+        }
+
+        /// <summary>
+        /// Get Handle size
+        /// </summary>
+        /// <param name="position">Position</param>
+        /// <returns>Handle size</returns>
+        public float GetHandleSize(Vector3 position)
+        {
+            var scale = HasCurrentCamera() ? 0.01f : 0.05f;
+            return HandleUtility.GetHandleSize(position) * scale;
+        }
+
+        /// <summary>
+        /// Helper to measure distance to segment describes 2 points
+        /// </summary>
+        /// <param name="p1">First point</param>
+        /// <param name="p2">Seconde point</param>
+        /// <returns>Distance between p1 and p2</returns>
+        public float DistanceToSegment(Vector3 p1, Vector3 p2)
+        {
+            p1 = HandleUtility.WorldToGUIPoint(p1);
+            p2 = HandleUtility.WorldToGUIPoint(p2);
+
+            return HandleUtility.DistancePointToLineSegment(Event.current.mousePosition, p1, p2);
+        }
+
+        /// <summary>
+        /// Helper to measure distance to circle
+        /// </summary>
+        /// <param name="center">Center of the circle</param>
+        /// <param name="radius">Radius of the circle</param>
+        /// <returns>Distance to circle</returns>
+        public float DistanceToCircle(Vector3 center, float radius)
```

Comment:
There is several function like this one that only retarget entry point.

Please, we already have a hell with GUI, EditorGUI, GUILayout, EditorGUILayout, GUIUtility, EsitorGUIUtility, EditorGUILayoutUtility for drawers. I don't think it is a good idea to repeat this for Handles. Handles are either in Handle, HandleUtility or any handle specialization SomethingHandle. Please don't add a new way to do things that already exist...